libfreerdp/core/gateway/http.c

```diff
@@ -130,7 +133,14 @@ static BOOL strings_equals_nocase(const void* obj1, const void* obj2)
 
 HttpContext* http_context_new(void)
 {
-	return (HttpContext*)calloc(1, sizeof(HttpContext));
+	HttpContext* context = (HttpContext*)calloc(1, sizeof(HttpContext));
+	if (context)
+	{
+		context->cookies = ListDictionary_New(FALSE);
+		ListDictionary_KeyObject(context->cookies)->fnObjectFree = free;
```

Comment:
might be a good idea to also set a new function, so you can remove the `strdup` below

Reply:
ok, I added `ListDictionary_KeyObject(context->cookies)->fnObjectNew = (OBJECT_NEW_FN)_strdup`

If I undestood it right ListDictionary_SetItemValue/ListDictionary_Add should make a copy of the string.

The compiler does however say `expected ‘void *’ but argument is of type ‘const char *’`
is it safe that I remove the const or are there dedicated function for that case?

Reply:
actually I don't see any use of fnObjectNew in ListDictionary.c.
are you sure this works? only fnObjectFree is used